### PR TITLE
analyze the transactional anomalies described by P. Baillis et al.

### PR DESCRIPTION
**Pull request by ikatkov**
More unit tests

### CODE REVIEW

File: hbase-client/src/test/java/org/apache/omid/transaction/TestBaillisAnomaliesWithTXs.java [hunk 1]
```diff
@@ -0,0 +1,597 @@
...
+            LOG.info("RESSS {}", res);
+            LOG.info("Deleting row id {} with value {}", Bytes.toString(res.getRow()),
+                     Bytes.toInt(res.getValue(famName, colName)));
+            Delete delete20 = new Delete(res.getRow());
+            txTable.delete(tx2, delete20);
+            res = tx2Scanner.next();
+            count20++;
+        }
+        assertEquals(count20, 1);
+//        int count20 = 0;
```
Comment (francisco-perez-sorrosal) [inline hbase-client/src/test/java/org/apache/omid/transaction/TestBaillisAnomaliesWithTXs.java]: Dead code

File: hbase-client/src/test/java/org/apache/omid/transaction/TestBaillisAnomaliesWithTXs.java [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+        updateRow4Tx2.add(famName, colName, Bytes.toBytes("42"));
+        txTable.put(tx2, updateRow4Tx2);
+
+        // 5) commit; -- T1
+        tm.commit(tx1);
+
+        // 6) commit; -- T2
+        tm.commit(tx2);
+
+        // 7) select * from test where value % 3 = 0; -- Either. Returns 3 => 30, 4 => 42
```
Comment (francisco-perez-sorrosal) [inline hbase-client/src/test/java/org/apache/omid/transaction/TestBaillisAnomaliesWithTXs.java]: Missing checks here

Comment (ikatkov) [inline hbase-client/src/test/java/org/apache/omid/transaction/TestBaillisAnomaliesWithTXs.java]: It does not fail with exception  - that is the check

File: hbase-client/src/test/java/org/apache/omid/transaction/TestBaillisAnomaliesWithTXs.java (hunk 1 above)
Comment (ikatkov) [inline hbase-client/src/test/java/org/apache/omid/transaction/TestBaillisAnomaliesWithTXs.java]: fixed

File: hbase-client/src/test/java/org/apache/omid/transaction/TestBaillisAnomaliesWithTXs.java [hunk 3]
```diff
@@ -0,0 +1,597 @@
...
+        TransactionManager tm = newTransactionManager(context);
+        TTable txTable = new TTable(hbaseConf, TEST_TABLE);
+        Transaction tx1 = tm.begin();
+        Transaction tx2 = tm.begin();
+
+        Scan rowId1Scan = new Scan(rowId1, rowId1);
+        rowId1Scan.addColumn(famName, colName);
+
+        // 1) select * from test where id = 1; -- T1
+        assertNumberOfRows(txTable, tx1, 2, new Scan());
```
Comment (francisco-perez-sorrosal) [inline hbase-client/src/test/java/org/apache/omid/transaction/TestBaillisAnomaliesWithTXs.java]: Should be:
   assertNumberOfRows(txTable, tx1, 1, rowId1Scan);

File: hbase-client/src/test/java/org/apache/omid/transaction/TestBaillisAnomaliesWithTXs.java [hunk 4]
```diff
@@ -0,0 +1,597 @@
...
+        try {
+            tm.commit(tx1);
+            fail("Should be aborted");
+        } catch (RollbackException e) {
+            // Expected
+        }
+
+    }
+
+    private void assertNumberOfRows(TTable txTable, Transaction tx2, int maxCount, Scan scan) throws IOException {
```
Comment (francisco-perez-sorrosal) [inline hbase-client/src/test/java/org/apache/omid/transaction/TestBaillisAnomaliesWithTXs.java]: Move to the end

Comment (ikatkov) [inline hbase-client/src/test/java/org/apache/omid/transaction/TestBaillisAnomaliesWithTXs.java]: done

**Pull request merged**